src/draw/vg_lite/lv_draw_vg_lite_vector.c

```diff
@@ -194,14 +194,15 @@ static void task_draw_cb(void * ctx, const lv_vector_path_t * path, const lv_vec
                 vg_lite_gradient_spreadmode_t spreadmode = lv_spread_to_vg(dsc->fill_dsc.gradient.spread);
                 LV_UNUSED(spreadmode);
 
-                lv_matrix_t m = dsc->matrix;
-                lv_matrix_translate(&m, min_x, min_y);
-                lv_matrix_multiply(&m, &dsc->fill_dsc.matrix);
+                if(style == LV_VECTOR_GRADIENT_STYLE_LINEAR) {
+                    vg_lite_matrix_t grad_matrix, fill_matrix;
+                    lv_area_t grad_area = { (int32_t)min_x, (int32_t)min_y, (int32_t)max_x, (int32_t)max_y};
+                    lv_vg_lite_grad_area_to_matrix(&grad_matrix, &grad_area, dsc->fill_dsc.gradient.grad.dir);
```

Comment:
`dsc->fill_dsc.gradient.grad.dir` -> `LV_GRAD_DIR_HOR`, use `fill_dsc.matrix` to control rotation uniformly.